build/cli-builder.sh

```diff
@@ -15,26 +13,31 @@ while IFS=, read -r git_url build_cmd build_dir; do
 
   git_repo=${git_url##*/}
 
-  cd "${REMOTE_SOURCES_DIR}/${git_repo}/${REMOTE_SOURCES_SUBDIR}"
+  cd "${SCRIPT_DIR}/../external/${git_repo}"
+  echo "=="
 
-  # Source relevant cachito.env for downstream builds
-  cachito_path="${REMOTE_SOURCES_DIR}/${git_repo}"/cachito.env
-  if [[ -f "${cachito_path}" ]]; then
-    source "${cachito_path}"
+  # Set branch using .gitmodules config
+  parent_gitmodules="${SCRIPT_DIR}/../.gitmodules"
+  if [[ -f "${parent_gitmodules}" ]]; then
+    submodule_branch=$(git config --file "${parent_gitmodules}" --get "submodule.${git_repo}.branch" 2>/dev/null || echo "")
+    if [[ -n "${submodule_branch}" ]]; then
+      echo "* Creating branch '${submodule_branch}' for version generation"
+      git checkout -b "${submodule_branch}" 2>/dev/null || echo "* Branch '${submodule_branch}' already exists or cannot be created"
+    fi
   fi
 
-  # Set branch using downstream CI_UPSTREAM_BRANCH variable
-  # ref: https://cpaas.pages.redhat.com/documentation/users/midstream/generating_files_from_templates.html#_environment_variables
-  if [[ -z "$(git branch --show-current)" ]] && [[ -n "${CI_UPSTREAM_BRANCH}" ]]; then
-    git checkout -b ${CI_UPSTREAM_BRANCH}
```

Comment:
For the sake of history, here was the reference in CPaaS:

```dockerfile
    && CI_UPSTREAM_BRANCH=${CI_UPSTREAM_BRANCH} GOFLAGS="-p=4" GOTAGS=${BUILD_TAGS} make build-and-package
```
ref: https://gitlab.cee.redhat.com/acm-cpaas-midstream/acm-cli/-/blame/rhacm-2.12-rhel-9/distgit/containers/acm-cli/Dockerfile.in?ref_type=heads#L27